usb/device/USBCDC_ECM/USBCDC_ECM.h

```diff
@@ -0,0 +1,209 @@
+    * Called when USB changes state
+    *
+    * @param new_state The new state of the USBDevice
+    *
+    * Warning: Called in ISR context
+    */
+    virtual void callback_state_change(DeviceState new_state);
+
+    /*
+    * This is used to handle extensions to standard requests
+    * and class specific requests with a data phase
+    */
+    virtual void callback_request_xfer_done(const setup_packet_t *setup, bool aborted);
+
+    /*
+    * Called by USBDevice layer. Set configuration of the device.
+    * For instance, you can add all endpoints that you need on this function.
+    *
+    * @param configuration Number of the configuration
+    * @returns true if class handles this request
+    */
+    virtual void callback_set_configuration(uint8_t configuration);
+
+    /*
+    * Called by USBDevice layer in response to set_interface.
+    *
+    * Upon reception of this command endpoints of any previous interface
+    * if any must be removed with endpoint_remove and new endpoint added with
+    * endpoint_add.
+    *
+    * @param configuration Number of the configuration
+    *
+    * Warning: Called in ISR context
+    */
+    virtual void callback_set_interface(uint16_t interface, uint8_t alternate);
+
+    /*
+    * Get device descriptor.
+    *
+    * @returns pointer to the device descriptor
+    */
+    virtual const uint8_t *device_desc();
+
+    /*
+    * Get string product descriptor
+    *
+    * @returns pointer to the string product descriptor
+    */
+    virtual const uint8_t *string_iproduct_desc();
+
+    /*
+    * Get string configuration descriptor
+    *
+    * @returns pointer to the string configuration descriptor
+    */
+    virtual const uint8_t *string_iconfiguration_desc();
+
+    /*
+    * Get string serial descriptor
+    *
+    * @returns pointer to the string serial descriptor
+    */
+    virtual const uint8_t *string_iserial_desc();
+
+    /*
+    * Get configuration descriptor
+    *
+    * @returns pointer to the configuration descriptor
+    */
+    virtual const uint8_t *configuration_desc(uint8_t index);
+
+    /*
+    * This is used to handle extensions to standard requests
+    * and class specific requests
+    */
+    virtual void callback_request(const setup_packet_t *setup);
+
+    /*
+    * Called by USBDevice layer on bus reset.
+    *
+    * complete_reset must be called after
+    * the device is fully reset.
+    *
+    * Warning: Called in ISR context
+    */
+    virtual void callback_reset();
+
+    uint8_t deviceDescriptor[18];
+
+private:
+
+    usb_ep_t _int_in;
+    usb_ep_t _bulk_in;
+    usb_ep_t _bulk_out;
+
+    uint8_t _bulk_buf[MAX_PACKET_SIZE_BULK];
+    uint32_t _bulk_buf_size;
+
+    rtos::EventFlags _flags;
+    rtos::Mutex _write_mutex;
+
+    void _init();
+    void _int_callback();
+    void _bulk_in_callback();
+    void _bulk_out_callback();
+    bool _notify_network_connection(uint8_t value);
+    bool _notify_connection_speed_change(uint32_t up, uint32_t down);
```

Comment:
`_notify_network_connection` and `_notify_connection_speed_change` are private and I don't see them being used anywhere in this class. Is this dead code?